core/iwasm/libraries/lib-wasi-threads/lib_wasi_threads_wrapper.c

```diff
@@ -122,6 +162,7 @@ thread_spawn_wrapper(wasm_exec_env_t exec_env, uint32 start_arg)
                                      thread_start_arg);
     if (ret != 0) {
         os_mutex_unlock(&exec_env->wait_lock);
+        deallocate_thread_id(thread_id);
```

Comment:
Maybe move it under the thread_spawn_fail label in case we'll have more fail branches after this one in the future? And even if not, I think it makes sense to have error handling in one place.